tiledb/sm/cpp_api/query.h

```diff
@@ -420,7 +420,9 @@ class Query {
              schema_.attribute(attr_name).cell_val_num() == TILEDB_VAR_NUM) ||
             (schema_.domain().has_dimension(attr_name) &&
              schema_.domain().dimension(attr_name).cell_val_num() ==
-                 TILEDB_VAR_NUM)));
+                 TILEDB_VAR_NUM) ||
+            (schema_.has_label(attr_name) &&
+             schema_.label(attr_name).label_cell_val_num() == TILEDB_VAR_NUM)));
```

Comment:
In the past, I've added duplicate functions in the experimental C++ API (see `get_data_buffer` in `query_experimental.h`). You might want to do the same thing here to keep experimental and non-experimental API separate.

Reply:
Thanks for the review! I think I hit all your points here. I added an experimental API for estimating results on the data query. I started with just `est_result_size_var` but if this looks good I can do the same for other variants of est_result_size.

Reply:
Adding the uint8 flag works. You can't have default parameters in C, but other than that it looks good.